src/lavinmq/http/controller/consumers.cr

```diff
@@ -32,31 +32,49 @@ module LavinMQ
           with_vhost(context, params) do |vhost|
             user = user(context)
             refuse_unless_management(context, user, vhost)
-            consumer_tag = URI.decode_www_form(params["consumer_tag"])
-            conn_id = URI.decode_www_form(params["connection"])
-            ch_id = URI.decode_www_form(params["channel"]).to_i
-            connection = connections(user).find { |conn| conn.vhost.name == vhost && conn.name == conn_id }
-            unless connection
-              context.response.status_code = 404
-              break
-            end
-            channel = connection.channels[ch_id]?
-            unless channel
+            consumer = find_consumer_from_params(user, vhost, params)
+            unless consumer
               context.response.status_code = 404
               break
             end
-            consumer = channel.consumers.find(&.tag.==(consumer_tag))
+            consumer.cancel
+            context.response.status_code = 204
+          end
+          context
+        end
+
+        put "/api/consumers/:vhost/:connection/:channel/:consumer_tag" do |context, params|
+          with_vhost(context, params) do |vhost|
+            user = user(context)
+            refuse_unless_management(context, user, vhost)
+            consumer = find_consumer_from_params(user, vhost, params)
             unless consumer
               context.response.status_code = 404
               break
             end
-            consumer.cancel
+            body = parse_body(context)
+            if prefetch = body["prefetch"]?.try(&.as_i?)
+              prefetch = 0 if prefetch < 0
+              prefetch = UInt16::MAX if prefetch > UInt16::MAX
```

Comment:
Should we abort with an error message instead if prefetch is `< 0` or `> UInt16::MAX` ? I think that's easier to understand for a user